src/components/organisms/WithNavigationBar/WithNavigationBar.tsx

```diff
@@ -16,17 +31,71 @@ export const WithNavigationBar: React.FunctionComponent<PropsType> = ({
   redirectionUrl,
   fullscreen,
   children,
-  hasBackButton,
-}) => (
-  <>
-    {/* @debt remove backButton from Navbar */}
-    <NavBar redirectionUrl={redirectionUrl} hasBackButton={hasBackButton} />
-    <div className={`navbar-margin ${fullscreen ? "fullscreen" : ""}`}>
-      {children}
+}) => {
+  const { user } = useUser();
+  const venueId = useVenueId();
+  const venue = useSelector(currentVenueSelectorData);
```

Comment:
I feel like we probably shouldn't just be using a selector here? We have hooks that wrap this for us. I would probably suggest `useConnectCurrentVenueNG` since it allows us to pass `venueId` to it directly.

Reply:
useConnectCurrentVenueNG returns the first venue in the collection when venueId is undefined or empty string, IMO this is wrong and in this case it doesn't work for me.

Reply:
That sounds like a bug in the hook? Probably better to fix that and then follow pruriens standard patterns?

Using the raw selectors like this, where are you connecting them to the data?

Reply:
@denisdimitrov Seems you haven't replied to this at all yet?

Reply:
@denisdimitrov Seems you haven't replied to this at all yet?